### ISSUE TITLE
wooden support rendering bug over crossing diagonal track

### ISSUE THREAD

**Issue by cheweytoo** [OpenRCT2/OpenRCT2#22617]
### Operating System

Ubuntu Linux 22.04.4

### OpenRCT2 build

develop 4cfc173, Linux AppImage

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This bug is a graphical glitch or error

### Describe the issue

A diagonal track crossing directly under a sloped track with wooden supports causes the supports to be drawn over the track:

![diagnoal_track_bug](https://github.com/user-attachments/assets/c74d9d5b-3706-41f5-be68-c440a17fb188)

This seems to only happen if the diagonal track itself has wooden supports too.

### Steps to reproduce

- build sloped track with wooden supports
- cross directly under it with diagonal track

See attached park.

### Attachments

[diagonal_track_bug.zip](https://github.com/user-attachments/files/16730781/diagonal_track_bug.zip)

**Comment by Gymnasiast** [comment]
I seem to remember this issue. Could you track down the release where this started happening? (You don’t need to bisect to build level necessarily, just release.)

**Comment by cheweytoo** [comment]
I didn't realise it was recent. I'll do some bisecting with the available dev builds.

**Comment by cheweytoo** [comment]
It's present in the oldest available dev build already unfortunately.

Going through the releases: Same issue in release 0.2.4(!), the oldest one with an AppImage build.

This smells a bit like an original bug to me now.

**Comment by Gymnasiast** [comment]
> I didn't realise it was recent. I'll do some bisecting with the available dev builds.

No, I meant it was old (as in: "I remember seeing this at one point").

If it’s already in v0.2.4, there is little point going further back. I’ll have to dive into the drawing code. Whoop de doo.

**Comment by cheweytoo** [comment]
> I’ll have to dive into the drawing code. Whoop de doo.

Good luck! :-}
